test-framework/keycloak-server/src/main/java/io/quarkus/test/keycloak/server/KeycloakContainer.java

```diff
@@ -0,0 +1,98 @@
+package io.quarkus.test.keycloak.server;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+
+import io.quarkus.runtime.configuration.ConfigurationException;
+import io.restassured.RestAssured;
+
+public class KeycloakContainer extends GenericContainer<KeycloakContainer> {
```

Comment:
Don't we have a dev service?

Reply:
In the `kafka-oauth-keycloak` IT we need to configure the kafka instance and the app with keycloak config. That's why we had an ad-hoc Keycloak test container – and the config was significantly complex, I simplified it in this change.

The `quarkus-test-keycloak-server` is recommended only when there is a reason not to use the Keycloak dev service for testing.

Reply:
@ozangunalp The refactoring looks good, and it is sound in any case, but Dev Service can be configured with a custom realm path, the Keycloak test factory is not really recommended any more, but indeed, quite a few OIDC tests still depend on it to test against legacy Keycloak distributions.

When you get a chance, please check what is missing from DevServicesConfig and may be we can enhance something, please open issues if you'll identify what can be. For example, I'm planning to support copying various test resources to the container to migrate KeycloakAuthorization tests to Dev Service